DOKS-automatic-node-repair/README.md

```diff
@@ -1 +1,402 @@
+    basicnp-8k4es   Ready    <none>   2m16s   v1.21.2
+    ```
+    ### Note:
+
+      If for some reason the `kubectl` context was not set properly we can always fix it via:
+      
+      ```bash
+      doctl k8s cluster kubeconfig save "$CLUSTER_NAME"
+      ```
+
+## Digital Mobius
+
+### Overview
+
+[Digital Mobius](https://github.com/Qovery/digital-mobius) is an open source project written in the `Go` language which can automatically recycle unhealthy nodes from a `DigitalOcean Kubernetes` cluster (aka `DOKS`).
+
+### How it Works
+
+The logic inside the application is watching for `DOKS` cluster nodes that are in an unhealthy state at a regular interval specified by the user. As described in the official `Kubernetes` documentation, a `node` is considered to be `unhealthy` if the [Node Condition](https://kubernetes.io/docs/concepts/architecture/nodes/#condition) is set to `Ready` and the status to `False` or `Unknown`. `Digital Mobius` looks for the `Unknown` status which is a result of the `master node` not being able to communicate with the worker node via the `kubelet`. It also provides a user configurable flag that specifies how much time has to pass before an unhealthy node needs to be recycled. If all the conditions enumerated earlier are met then the affected node(s) will be re-created. It does so via a call to the [Delete Kubernetes Node](https://docs.digitalocean.com/reference/api/api-reference/#operation/delete_kubernetes_node) `REST API` provided by `DigitalOcean`.
+
+Below is a simplified diagram showing how `Digital Mobius` checks the worker node(s) state:
+
+![Digital Mobius Flow](content/img/digital-mobius-flow.png)
+
+### Configuration Overview
+
+In order for the application to work it needs a set of `environment variables` to be configured and available as seen in the [Helm Chart](https://github.com/Qovery/digital-mobius/blob/main/charts/Digital-Mobius/values.yaml) values file (explanations inline):
+
+```
+LOG_LEVEL: "info"
+DELAY_NODE_CREATION: "10m"                                  # A worker node gets recycled after being unhealthy for this period of time
+DIGITAL_OCEAN_TOKEN: "<your_digital_ocean_api_token>"       # Personal DO API token value
+DIGITAL_OCEAN_CLUSTER_ID: "<your_digital_ocean_cluster_id>" # DOKS cluster ID that needs to be monitored
+```
+
+The most important values that we need to provide here is the DigitalOcean `API token`, the `Cluster ID` and the `Node Creation Delay` time interval which can be expressed in seconds or minutes using the appropriate suffix - e.g.: `10s`, `10m`, etc.
+
+**Note:**
+
+Having a value that's too low for the `DELAY_NODE_CREATION` option will interfere with the time interval needed for a node to become ready and available after it gets recycled. From real world tests it can take several minutes to complete or even more, so please choose a value that's appropriate. A good starting point is `10m` - hence the value used in this tutorial.
```

Comment:
```suggestion
Having a value that's too low for `DELAY_NODE_CREATION` will interfere with the time interval needed for a node to become ready and available after it gets recycled. From real world situations, it can take several minutes to complete or even more, so please choose a value that's appropriate. A good starting point is `10m` - hence the value used in this tutorial.
```